pymatgen/io/openmm/parameterizer.py

```diff
@@ -0,0 +1,441 @@
+            The parameterizer type to assign to the force fields (if parameterizer_assignment is ParameterizerAssignment.EXPLICIT).
+        parameterizer_assignment : ParameterizerAssignment
+            The method of parameterizer assignment to use (INFERRED, EXPLICIT, or DEFAULT).
+
+        Raises
+        ------
+        AssertionError
+            If the parameterizer assignment type or parameterizer type (if specified) is not of the correct type.
+            If multiple parameterizer types are being used for the force fields.
+            If an explicit parameterizer type is required but not provided.
+
+        Returns
+        -------
+        None
+        """
+        prev_parameterizer_type = ParameterizerType.DEFAULT_PARAMETERIZER
+       
+        for i in range(len(force_fields)):
+            ff = force_fields[i]
+
+            assert type(parameterizer_assignment) == ParameterizerAssignment, INVALID_ASSIGNMENT_TYPE
+            assignment = parameterizer_assignment
+            
+            if parameterizer_type:
+                assert type(parameterizer_type) == ParameterizerType, INVALID_PARAMETERIZER_TYPE
+
+            if assignment == ParameterizerAssignment.INFERRED:
+                self.infer_parameterizer_type(ff)
+            elif assignment == ParameterizerAssignment.EXPLICIT:
+                assert parameterizer_type != None, PARAMETERIZER_TYPE_REQUIRED
+                if(parameterizer_type != None):
+                    self.parameterizer_type = parameterizer_type
+            elif assignment == ParameterizerAssignment.DEFAULT:
+                self.parameterizer_type  = ParameterizerType.DEFAULT_PARAMETERIZER
+            else:
+                self.parameterizer_type  = ParameterizerType.DEFAULT_PARAMETERIZER
+
+            if i > 0:
+                assert self.parameterizer_type == prev_parameterizer_type, MULTIPLE_PARAMETERIZERS_NOT_SUPPORTED
+    
+    def assign_topology(self, topology:  tk.Topology):
+        """
+        Method that assigns a topology for the parameterizer based on the parameterizer type.
+
+        Parameters
+        ----------
+        topology : tk.Topology
+            The topology to assign.
+        
+        Returns
+        -------
+        None
+        """
+        if(self.parameterizer_type == ParameterizerType.OPENMM_PARAMETERIZER):
+            self.topology = topology.to_openmm()
+        else:
+            self.topology = topology
+    
+    def assert_valid_inputs(self):
+        """
+        Method that validates the inputs used to initialize the parameterizer.
+
+        Raises
+        ------
+        AssertionError
+            If the force field type is not supported by the parameterizer.
+            If the topology type is unexpected for the parameterizer type.
+            If custom force field files are being used with a non-OpenMM parameterizer.
+            If no custom force field files are provided when using custom force fields.
```

Comment:
How are we currently supporting arbitrary .xml files that aren't in the interchange, openmm, or openff lists?